File: experiments/polyglot/services/java/src/polyglot/IdentityService.java
package polyglot;

import com.sun.net.httpserver.HttpServer;
import dev.drim.azimuth.Azimuth;
import java.net.InetSocketAddress;
import java.nio.charset.StandardCharsets;

public final class IdentityService {
    private IdentityService() {}

    @Azimuth.Realizes(spec = "polyglot/identity", scenario = "java-identifies")
    public static String identity() {
        return "java";
    }

    public static void main(String[] arguments) throws Exception {
        int port = Integer.parseInt(System.getenv().getOrDefault("PORT", "8082"));
        HttpServer server = HttpServer.create(new InetSocketAddress(port), 0);
        server.createContext("/identity", exchange -> {
            byte[] body = (identity() + "\n").getBytes(StandardCharsets.UTF_8);
            exchange.sendResponseHeaders(200, body.length);
            exchange.getResponseBody().write(body);
            exchange.close();
        });
        server.start();
    }
}

File: experiments/polyglot/services/kotlin/src/main/kotlin/polyglot/IdentityService.kt
package polyglot

import com.sun.net.httpserver.HttpServer
import dev.drim.azimuth.Azimuth
import java.net.InetSocketAddress

object IdentityService {
    @JvmStatic
    @Azimuth.Realizes(spec = "polyglot/identity", scenario = "kotlin-identifies")
    fun identity(): String = "kotlin"

    @JvmStatic
    fun main(arguments: Array<String>) {
        val port = System.getenv("PORT")?.toInt() ?: 8083
        val server = HttpServer.create(InetSocketAddress(port), 0)
        server.createContext("/identity") { exchange ->
            val body = "${identity()}\n".toByteArray()
            exchange.sendResponseHeaders(200, body.size.toLong())
            exchange.responseBody.use { it.write(body) }
        }
        server.start()
    }
}
